src/dotnetTestExplorer.ts

```diff
@@ -111,28 +111,43 @@ export class DotnetTestExplorer implements TreeDataProvider<TestNode> {
         }
     }
 
+    private enableBuild(): string {
+        const dotnetBuildOption = Utility.getConfiguration().get<boolean>("build");
+        return dotnetBuildOption ? "" : "--no-build";
+    }
+
+    private enableRestore(): string {
+        const dotnetRestoreOption = Utility.getConfiguration().get<boolean>("restore");
+        return dotnetRestoreOption ? "" : "--no-restore";
+    }
+
+    private loadExclusion(name: string, value: string): string {
+        const exclusion = Utility.getConfiguration().get<string>(name);
+        return exclusion ? exclusion : value;
+    }
+
     private loadTestStrings(): Thenable<string[]> {
         this.evaluateTestDirectory();
 
-        let msBuildRootTestMsg = Utility.getConfiguration().get<string>("msbuildRootTestMsg");
-        msBuildRootTestMsg = msBuildRootTestMsg ? msBuildRootTestMsg : "The following Tests are available:";
+        const exclusions = [];
+
+        exclusions.push(this.loadExclusion("msbuildRootTestxUnitPrefix", "[xUnit.net"));

```

Comment:
Because selecting only one place to slice the text only works if the test projects are perfectly in order and are the last projects in the solution.

Otherwise, the `dotnet test -t` output is a pretty big mess.

We have to filter the text no matter what, but `filter` performs on the observable string array and is much more optimized than trying to run through the array multiple times.

This new method inspects the whole array only once, and drops unnecessary text so as to leave us with the tests no matter what the project order and arrangement, and leverages any memory, cache, and performance advantages that the underlying observable gives us.